### PR TITLE
docs - add content for kerberos constrained delegation support

### PR DESCRIPTION
**Pull request by lisakowen**
update the docs for pxf support of kerberos constrained delegation. in this PR:
- update servers topic to include general info about the new server-specific pxf.service.kerberos.constrained-delegationn property
- add a new graphic for one of the new use cases
- many updates to the existing "configuring pxf for secure hadoop" topic:
    - intro topic updates
    - add section for "accessing hadoop using constrained delegation".  includes addtl prereqs, enable/disable procedure, and new scenarios
    - restructure the ad- and mit- specific KDC instructions a bit;  add eda new "completing the pxf config" topic that both ad/mit procedures now reference
- impersonation topic - misc edits to clarify some info

i know this new info isn't ideally placed.  i need to re-work how some of this info is presented, will tackle that after release.

doc review site links (behind vpn):
- configuring servers topic :  https://docs-lisa-pxf-kcd.sc2-04-pcf1-apps.oc.vmware.com/pxf/6-3/using/cfg_server.html#pxf-site
- hadoop impersonation topic:  https://docs-lisa-pxf-kcd.sc2-04-pcf1-apps.oc.vmware.com/pxf/6-3/using/pxfuserimpers.html
- hadoop kerberos topic:  https://docs-lisa-pxf-kcd.sc2-04-pcf1-apps.oc.vmware.com/pxf/6-3/using/pxf_kerbhdfs.html

### CODE REVIEW

File: docs/content/pxf_kerbhdfs.html.md.erb [hunk 1]
```diff
@@ -92,7 +94,91 @@ The following table identifies the `pxf.service.user.impersonation` and `pxf.ser
...
+
+    ``` shell
+    gpadmin@gpmaster$ pxf cluster sync
+    ```
+
+#### <a id="kcd_gp"></a>Accessing Hadoop On Behalf of the Greenplum User
+
+In this configuration, PXF uses Kerberos constrained delegation to request and obtain a ticket on behalf of the Greenplum user, and uses the ticket to access Hadoop.
+
+<img alt="Accessing Hadoop using Kerberos Constrained Delegation" src="graphics/impersonation_cases/impersonation-case-kcd.png" class="image" />
```
Comment (bradfordb-vmware) [inline docs/content/pxf_kerbhdfs.html.md.erb:162]: I can't seem to put comments on the image, so I'll comment here -- instead of `KDC / LDAP`, would we want to say `AD / LDAP`?

File: docs/content/pxf_kerbhdfs.html.md.erb [hunk 2]
```diff
@@ -152,24 +238,16 @@ When you configure PXF for secure HDFS using an AD Kerberos KDC server, you will
...
 
-5. Synchronize the PXF configuration to your Greenplum Database cluster and restart PXF:
-
-    ``` shell
-    gpadmin@master$ pxf cluster sync
-    gpadmin@master$ pxf cluster restart
-    ```
-
-6. Step 7 does not synchronize the keytabs in `$PXF_BASE`. You must distribute the keytab file to `$PXF_BASE/keytabs/`. Locate the keytab file, copy the file to the `$PXF_BASE` runtime configuration directory, and set required permissions. For example:
+6. Synchronize the keytabs in `$PXF_BASE`. You must distribute the keytab file to `$PXF_BASE/keytabs/`. Locate the keytab file, copy the file to the `$PXF_BASE` runtime configuration directory, and set required permissions. For example:
```
Comment (bradfordb-vmware) [inline docs/content/pxf_kerbhdfs.html.md.erb:243]: Nit

```suggestion
5. Synchronize the keytabs in `$PXF_BASE`. You must distribute the keytab file to `$PXF_BASE/keytabs/`. Locate the keytab file, copy the file to the `$PXF_BASE` runtime configuration directory, and set required permissions. For example:
```

File: docs/content/pxf_kerbhdfs.html.md.erb [hunk 3]
```diff
@@ -274,22 +352,30 @@ When you configure PXF for secure HDFS using an MIT Kerberos KDC server, you wil
     
     PXF automatically replaces ` _HOST` with the FQDN of the host.
 
-6. If you want to access Hadoop as the Greenplum Database user:
+6. [Complete the PXF Configuration](#proc_complete) based on your chosen Hadoop access scenario.
+
+### <a id="proc_scenario"></a>Completing the PXF Configuration
```
Comment (bradfordb-vmware) [inline docs/content/pxf_kerbhdfs.html.md.erb:357]: Should this anchor id be `proc_complete`?

File: docs/content/pxfuserimpers.html.md.erb [hunk 4]
```diff
@@ -2,11 +2,11 @@
 title: Configuring the Hadoop User, User Impersonation, and Proxying
 ---
 
-PXF accesses Hadoop services on behalf of Greenplum Database end users.
+PXF accesses Hadoop services on behalf of Greenplum Database end users. Impersonation is a way to present a Greenplum end user identity to a remote system. You can achieve this with PXF via proxying. When the Hadoop service is secured with Kerberos, you also have the option of impersonation using Kerberos constrained delegation.
```
Comment (denalex) [inline docs/content/pxfuserimpers.html.md.erb:5]: `via proxying` is not very clear, but I don't have a better wording.

File: docs/content/pxfuserimpers.html.md.erb [hunk 5]
```diff
@@ -2,11 +2,11 @@
...
 
-When user impersonation is disabled, PXF executes all Hadoop service requests as the PXF process owner (usually `gpadmin`) or the Hadoop user identity that you specify. This behavior provides no means to control access to Hadoop services for different Greenplum Database users. It requires that this user have access to all files and directories in HDFS, and all tables in Hive and HBase that are referenced in PXF external table definitions.
+When user impersonation for a non-secured Hadoop cluster is disabled, PXF executes all Hadoop service requests as the PXF process owner (usually `gpadmin`) or the Hadoop user identity that you specify. This behavior provides no means to control access to Hadoop services for different Greenplum Database users. It requires that this user have access to all files and directories in HDFS, and all tables in Hive and HBase that are referenced in PXF external table definitions.
```
Comment (denalex) [inline docs/content/pxfuserimpers.html.md.erb:9]: this edit now made this section about non-secure clusters, not sure why this was needed.

File: docs/content/pxfuserimpers.html.md.erb [hunk 6]
```diff
@@ -106,7 +106,7 @@ Perform the following procedure to configure the Hadoop user:
     gpadmin@gpmaster$ cp $PXF_HOME/templates/pxf-site.xml .
     ```
 
-5. Open the `pxf-site.xml` file in the editor of your choice, and configure the Hadoop user name. When impersonation is disabled, this name identifies the Hadoop user identity that PXF will use to access the Hadoop system. When user impersonation is enabled, this name identifies the PXF proxy Hadoop user. For example, if you want to access Hadoop as the user `hdfsuser1`, uncomment the property and set it as follows:
+5. Open the `pxf-site.xml` file in the editor of your choice, and configure the Hadoop user name. For example, if you want to access Hadoop as the user `hdfsuser1`, uncomment the property and set it as follows:
```
Comment (denalex) [inline docs/content/pxfuserimpers.html.md.erb:109]: Please do not delete that sentence. This is second use of serviceUser property when impersonation IS enabled for non-secure clusters. Maybe it needs to be re-phrased to distinguish between secure/non-secure, but remember, this is the source of truth:
```
<description>

            Uncomment and set the proper value only if:

            - user impersonation is enabled and you want to use the specified
              user as a proxy on the unsecured Hadoop clusters. This is useful
              when a proxy user has already been configured on the Hadoop side,
              and you don't want to add gpadmin (the default) as a proxy user.

            - user impersonation is disabled and you want queries from all
              Greenplum users to appear on the Hadoop side as coming from the
              specified user.

        </description>
```

File: docs/content/pxfuserimpers.html.md.erb [hunk 7]
```diff
@@ -158,8 +158,6 @@ PXF user impersonation is enabled by default for Hadoop servers. You can configu
     </property>
     ```
 
-3. If you enabled user impersonation, you must configure Hadoop proxying as described in [Configure Hadoop Proxying](#hadoop). You must also configure [Hive User Impersonation](#hive) and [HBase User Impersonation](#hbase) if you plan to use those services.
```
Comment (denalex) [inline docs/content/pxfuserimpers.html.md.erb:161]: why was this deleted ?

File: docs/content/pxfuserimpers.html.md.erb [hunk 8]
```diff
@@ -171,11 +169,11 @@ PXF user impersonation is enabled by default for Hadoop servers. You can configu
 
 ## <a id="hadoop"></a>Configure Hadoop Proxying
 
-When PXF user impersonation is enabled for a Hadoop server configuration, you must configure Hadoop to permit PXF to proxy Greenplum users. This configuration involves setting certain `hadoop.proxyuser.*` properties. Follow these steps to set up PXF Hadoop proxy users:
+When PXF user impersonation is enabled in the PXF server configuration of a non-secured Hadoop server configuration, you must configure Hadoop to permit PXF to proxy Greenplum users. This configuration involves setting certain `hadoop.proxyuser.*` properties. Follow these steps to set up PXF Hadoop proxy users:
```
Comment (denalex) [inline docs/content/pxfuserimpers.html.md.erb:172]: why only non-secured ?

File: docs/content/pxfuserimpers.html.md.erb [hunk 9]
```diff
@@ -171,11 +169,11 @@ PXF user impersonation is enabled by default for Hadoop servers. You can configu
...
+2. Set the property `hadoop.proxyuser.<name>.hosts` to specify the list of PXF host names from which proxy requests are permitted. Substitute the PXF proxy Hadoop user for `<name>`. The PXF proxy Hadoop user is the `pxf.service.user.name` that you configured in the procedure above, or, if you are using Kerberos authentication to Hadoop without constrained delegation, the proxy user identity is the *primary* component of the Kerberos principal. If you have not explicitly configured `pxf.service.user.name`, the proxy user is the operating system user that started PXF. Provide multiple PXF host names in a comma-separated list. For example, if the PXF proxy user is named `hdfsuser2`:
```
Comment (denalex) [inline docs/content/pxfuserimpers.html.md.erb:176]: I am not sure we need this edit, configuring proxy be definition is for the case where KCD is not used. Maybe mention that fact in the beginning of the section.

File: docs/content/pxf_kerbhdfs.html.md.erb [hunk 10]
```diff
@@ -92,7 +94,91 @@ The following table identifies the `pxf.service.user.impersonation` and `pxf.ser
 with a &lt;custom> user impersonating Greenplum users. PXF requires that you
 impersonate Greenplum users using the Kerberos principal.
 
-## <a id="procedure"></a>Procedure
+### <a id="kcd"></a>Accessing Hadoop Using Kerberos Constrained Delegation
+
+Kerberos constrained delegation is a Windows Server feature that allows an administrator to specify trust boundaries that restrict the scope of where an application can act on behalf of a user. You may choose to configure PXF to use Kerberos constrained delegation when you want to manage HDFS access privileges in a directory service without the need to specify a proxy Hadoop user. Refer to the Microsoft [Service for User](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-sfu/bde93b0e-f3c9-4ddf-9f44-e1453be7af5a) (S4U) Kerberos protocol extension documentation for more information about Kerberos constrained delegation.
```
[the remark this answers is not in the document]
Comment (lisakowen) [reply docs/content/pxf_kerbhdfs.html.md.erb]: replaced with feature

File: docs/content/pxfuserimpers.html.md.erb (hunk 8 above)
Comment (lisakowen) [reply docs/content/pxfuserimpers.html.md.erb]: you're right, added "and kcd is disabled".

File: docs/content/pxfuserimpers.html.md.erb (hunk 6 above)
Comment (lisakowen) [reply docs/content/pxfuserimpers.html.md.erb]: i added the sentence back.

File: docs/content/pxf_kerbhdfs.html.md.erb [hunk 11]
```diff
@@ -53,13 +53,15 @@ In this configuration, PXF accesses Hadoop as the Greenplum user proxied
...
+This is the default configuration for a Hadoop PXF server.
+
 <img alt="Accessing Hadoop as the Greenplum User Proxied by the Kerberos Principal" src="graphics/impersonation_cases/impersonation-case-1-kerberos.png" class="image" />
 
 The following table identifies the `pxf.service.user.impersonation` and `pxf.service.user.name` settings, and the PXF and Hadoop configuration required for this use case:
 
 | Impersonation  | Service&nbsp;User |  PXF Configuration | Hadoop&nbsp;Configuration&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; |
 |----------------|-------------------|--------------------|-----------|
-|  true          | Kerberos principal | Perform the [Configuration Procedure](#procedure) in this topic. | Set the Kerberos principal as the Hadoop proxy user as described in [Configure Hadoop Proxying](pxfuserimpers.html#hadoop). |
+|  true          | Identity of the Kerberos principal | Perform the [Configuration Procedure](#procedure) in this topic. | Set the Kerberos principal as the Hadoop proxy user as described in [Configure Hadoop Proxying](pxfuserimpers.html#hadoop). |
```
[the remark this answers is not in the document]
Comment (lisakowen) [reply docs/content/pxf_kerbhdfs.html.md.erb:64]: i updated the para preceding the table to not use the property names (like in the scenarios that i added).  i'll take a look at this again when i complete all edits.

File: docs/content/pxfuserimpers.html.md.erb (hunk 7 above)
Comment (lisakowen) [reply docs/content/pxfuserimpers.html.md.erb:161]: i deleted this line because it is not strictly true when kerberos constrained delegation is enabled.

File: docs/content/pxf_kerbhdfs.html.md.erb (hunk 1 above)
Comment (lisakowen) [reply docs/content/pxf_kerbhdfs.html.md.erb]: i am happy to change this, but isn't AD more specific, where as KDC is more generic and could cover an IPA server?  (i might be totally be misunderstanding these terms ...)

File: docs/content/pxf_kerbhdfs.html.md.erb [hunk 12]
```diff
@@ -92,7 +94,91 @@ The following table identifies the `pxf.service.user.impersonation` and `pxf.ser
 with a &lt;custom> user impersonating Greenplum users. PXF requires that you
 impersonate Greenplum users using the Kerberos principal.
 
-## <a id="procedure"></a>Procedure
+### <a id="kcd"></a>Accessing Hadoop Using Kerberos Constrained Delegation
```
[the remark this answers is not in the document]
Comment (lisakowen) [reply docs/content/pxf_kerbhdfs.html.md.erb]: i attempted to do this.  please take a look at the commit or review site.

File: docs/content/pxf_kerbhdfs.html.md.erb [hunk 13]
```diff
@@ -92,7 +94,91 @@ The following table identifies the `pxf.service.user.impersonation` and `pxf.ser
...
+    </property>
+    ```
+
+1. Save the `pxf-site.xml` file and exit the editor.
+
+1. Use the `pxf cluster sync` command to synchronize the PXF Hadoop server configuration to your Greenplum Database cluster:
+
+    ``` shell
+    gpadmin@gpmaster$ pxf cluster sync
+    ```
```
[the remark this answers is not in the document]
Comment (lisakowen) [reply docs/content/pxf_kerbhdfs.html.md.erb]: yeah, i get this.  i will look into doing this refactor later.

File: docs/content/pxf_kerbhdfs.html.md.erb [hunk 14]
```diff
@@ -274,22 +352,30 @@ When you configure PXF for secure HDFS using an MIT Kerberos KDC server, you wil
...
+6. [Complete the PXF Configuration](#proc_complete) based on your chosen Hadoop access scenario.
+
+### <a id="proc_scenario"></a>Completing the PXF Configuration
 
-    1. Enable user impersonation as described in [Configure PXF User Impersonation](pxfuserimpers.html#pxf_cfg_impers).
-    2. Configure Hadoop proxying for the *primary* component of the Kerberos principal as described in [Configure Hadoop Proxying](pxfuserimpers.html#hadoop). For example, if your principal is `gpadmin/_HOST@EXAMPLE.COM`, configure proxying for the Hadoop user `gpadmin`.
+On the Greenplum Database master host, complete the configuration of the PXF server based on your chosen Hadoop access scenario:
 
-7. If you want to access Hadoop using the identity of the Kerberos principal, disable user impersonation as described in [Configure PXF User Impersonation](pxfuserimpers.html#pxf_cfg_impers).
+1. If you want to access Hadoop using Kerberos constrained delegation,
```
[the remark this answers is not in the document]
Comment (lisakowen) [reply docs/content/pxf_kerbhdfs.html.md.erb]: i reworked these steps in the procedure.

**Comment by lisakowen** [comment]
@denalex - thank you for your comments, they helped me understand the feature better.  the organization of the topic is much clearer now.

File: docs/content/pxf_kerbhdfs.html.md.erb [hunk 15]
```diff
@@ -4,23 +4,37 @@ title: Configuring PXF for Secure HDFS
...
+The identity with which PXF accesses a Kerberos-secured Hadoop depends on the settings of the following properties:
 
-You configure the impersonation setting and the Kerberos principal and keytab for a Hadoop server via the `pxf-site.xml` server-specific configuration file. Refer to [About the pxf-site.xml Configuration File](cfg_server.html#pxf-site) for more information about the configuration properties in this file.
+| Property       |  Description | Default Value |
+|----------------|--------------|---------------|
+| pxf.service.kerberos.principal | The PXF Kerberos principal name. | gpadmin/\_HOST@EXAMPLE.COM |
+| pxf.service.user.impersonation | Enables/disables Greenplum Database user impersonation on the remote system. | `true` |
+| pxf.service.kerberos.constrained-delegation | Enables/disables usage of Kerberos constrained delegation based on S4U Kerberos extensions. This option allows Hadoop administrators to avoid creating a proxy user configuration in Hadoop, instead requiring them to perform delegation configuration in and LDAP server. | `false` |
```
Comment (bradfordb-vmware) [inline docs/content/pxf_kerbhdfs.html.md.erb:15]:
```suggestion
| pxf.service.kerberos.constrained-delegation | Enables/disables usage of Kerberos constrained delegation based on S4U Kerberos extensions. This option allows Hadoop administrators to avoid creating a proxy user configuration in Hadoop, instead requiring them to perform delegation configuration in an Active Directory (AD) or Identity Policy Audit (IPA) server. | `false` |
```

File: docs/content/pxf_kerbhdfs.html.md.erb [hunk 16]
```diff
@@ -4,23 +4,37 @@ title: Configuring PXF for Secure HDFS
...
+**Note:** PXF supports simultaneous access to multiple Kerberos-secured Hadoop clusters.
+
+## <a id="kcd"></a>About Kerberos Constrained Delegation
+
+Kerberos constrained delegation is a feature that allows an administrator to specify trust boundaries that restrict the scope of where an application can act on behalf of a user. You may choose to configure PXF to use Kerberos constrained delegation when you want to manage user impersonation privileges in a directory service without the need to specify a proxy Hadoop user. Refer to the Microsoft [Service for User](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-sfu/bde93b0e-f3c9-4ddf-9f44-e1453be7af5a) (S4U) Kerberos protocol extension documentation for more information about Kerberos constrained delegation.
+
+When your Active Directory or Identity Policy Audit (IPA) Server is configured appropriately and you enable Kerberos constrained delegation for PXF, the PXF service requests and obtains a Kerberos ticket on behalf of the user, and uses the ticket to access the HDFS file system. PXF caches the ticket for one day.
```
Comment (bradfordb-vmware) [inline docs/content/pxf_kerbhdfs.html.md.erb:26]:
```suggestion
When your AD or IPA server is configured appropriately and you enable Kerberos constrained delegation for PXF, the PXF service requests and obtains a Kerberos ticket on behalf of the user, and uses the ticket to access the HDFS file system. PXF caches the ticket for one day.
```

File: docs/content/pxf_kerbhdfs.html.md.erb [hunk 17]
```diff
@@ -40,26 +54,48 @@ Before you configure PXF for access to a secure HDFS filesystem, ensure that you
     root@gphost$ yum install krb5-libs krb5-workstation
     ```
 
+Ensure that you meet these additional prerequisites when PXF uses Kerberos constrained delegation:
+
+- S4U is enabled in the Active Directory or IPA Server.
```
Comment (bradfordb-vmware) [inline docs/content/pxf_kerbhdfs.html.md.erb:59]:
```suggestion
- S4U is enabled in the AD or IPA server.
```

File: docs/content/pxf_kerbhdfs.html.md.erb [hunk 18]
```diff
@@ -40,26 +54,48 @@ Before you configure PXF for access to a secure HDFS filesystem, ensure that you
...
+
+- The Active Directory or IPA Server is configured to allow the PXF Kerberos principal to impersonate end users.
```
Comment (bradfordb-vmware) [inline docs/content/pxf_kerbhdfs.html.md.erb:61]:
```suggestion
- The AD or IPA server is configured to allow the PXF Kerberos principal to impersonate end users.
```

File: docs/content/pxf_kerbhdfs.html.md.erb [hunk 19]
```diff
@@ -40,26 +54,48 @@ Before you configure PXF for access to a secure HDFS filesystem, ensure that you
...
+
+In this configuration, PXF uses Kerberos constrained delegation to request and obtain a ticket on behalf of the Greenplum user, and uses the ticket to access Hadoop.
+
+<img alt="Accessing Hadoop using Kerberos Constrained Delegation" src="graphics/impersonation_cases/impersonation-case-kcd.png" class="image" />
+
+The following table identifies the impersonation and service user settings, and the PXF and directory service configuration required for this use case; no Hadoop configuration is required:
+
+| Impersonation  | Service&nbsp;User |  PXF Configuration | AD/IPA&nbsp;Config&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; |
+|----------------|-------------------|--------------------|-----------|
+|  true          | Greenplum user | Set up the PXF Kerberos principal, keytab files, and related settings in `pxf-site.xml` as described in the [Configuration Procedure](#procedure) in this topic, and [Enable Kerberos Constrained Delegation](#enable_kcd). | Configure LDAP to provide the PXF Kerberos principal with the delegation rights for the Greenplum end users. |
```
Comment (bradfordb-vmware) [inline docs/content/pxf_kerbhdfs.html.md.erb:98]:
```suggestion
|  true          | Greenplum user | Set up the PXF Kerberos principal, keytab files, and related settings in `pxf-site.xml` as described in the [Configuration Procedure](#procedure) in this topic, and [Enable Kerberos Constrained Delegation](#enable_kcd). | Configure AD or IPA to provide the PXF Kerberos principal with the delegation rights for the Greenplum end users. |
```

File: docs/content/pxf_kerbhdfs.html.md.erb [hunk 20]
```diff
@@ -274,22 +316,69 @@ When you configure PXF for secure HDFS using an MIT Kerberos KDC server, you wil
...
-    2. Configure Hadoop proxying for the *primary* component of the Kerberos principal as described in [Configure Hadoop Proxying](pxfuserimpers.html#hadoop). For example, if your principal is `gpadmin/_HOST@EXAMPLE.COM`, configure proxying for the Hadoop user `gpadmin`.
+On the Greenplum Database master host, complete the configuration of the PXF server based on your chosen Hadoop access scenario:
 
-7. If you want to access Hadoop using the identity of the Kerberos principal, disable user impersonation as described in [Configure PXF User Impersonation](pxfuserimpers.html#pxf_cfg_impers).
+1. If you want to access Hadoop as the Greenplum Database user:
 
-8. If you want to access Hadoop as a custom user:
+    1. Enable user impersonation as described in [Configure PXF User Impersonation](pxfuserimpers.html#pxf_cfg_impers) (this is the default setting).
+    1. *If you want to use Kerberos constrained delegation*, [enable](#enable_kdc) it for the server, and configure LDAP to provide the PXF Kerberos principal with the delegation rights for the Greenplum end users.
```
Comment (bradfordb-vmware) [inline docs/content/pxf_kerbhdfs.html.md.erb:328]:
```suggestion
    1. *If you want to use Kerberos constrained delegation*, [enable](#enable_kdc) it for the server, and configure AD or IPA to provide the PXF Kerberos principal with the delegation rights for the Greenplum end users.
```

File: docs/content/pxf_kerbhdfs.html.md.erb [hunk 21]
```diff
@@ -274,22 +316,69 @@ When you configure PXF for secure HDFS using an MIT Kerberos KDC server, you wil
...
+
+1. If you want to access Hadoop using the identity of the Kerberos principal, disable user impersonation as described in [Configure PXF User Impersonation](pxfuserimpers.html#pxf_cfg_impers).
+
+1. If you want to access Hadoop as a custom user:
 
     1. Disable user impersonation as described in [Configure PXF User Impersonation](pxfuserimpers.html#pxf_cfg_impers).
     2. Configure the custom user name as described in [Configure the Hadoop User](pxfuserimpers.html#pxf_cfg_user).
-    3. Configure Hadoop proxying for the *primary* component of the Kerberos principal as described in [Configure Hadoop Proxying](pxfuserimpers.html#hadoop). For example, if your principal is `gpadmin/_HOST@EXAMPLE.COM`, configure proxying for the Hadoop user `gpadmin`.
+    1. *If you want to use Kerberos constrained delegation*, [enable](#enable_kcd) it for the server, and configure LDAP to provide the PXF Kerberos principal with the delegation rights for the custom user.
```
Comment (bradfordb-vmware) [inline docs/content/pxf_kerbhdfs.html.md.erb:337]:
```suggestion
    1. *If you want to use Kerberos constrained delegation*, [enable](#enable_kcd) it for the server, and configure AD or IPA to provide the PXF Kerberos principal with the delegation rights for the custom user.
```

File: docs/content/pxfuserimpers.html.md.erb [hunk 22]
```diff
@@ -2,11 +2,11 @@
...
 
-When user impersonation is disabled, PXF executes all Hadoop service requests as the PXF process owner (usually `gpadmin`) or the Hadoop user identity that you specify. This behavior provides no means to control access to Hadoop services for different Greenplum Database users. It requires that this user have access to all files and directories in HDFS, and all tables in Hive and HBase that are referenced in PXF external table definitions.
+When user impersonation, PXF executes all Hadoop service requests as the PXF process owner (usually `gpadmin`) or the Hadoop user identity that you specify. This behavior provides no means to control access to Hadoop services for different Greenplum Database users. It requires that this user have access to all files and directories in HDFS, and all tables in Hive and HBase that are referenced in PXF external table definitions.
```
Comment (bradfordb-vmware) [inline docs/content/pxfuserimpers.html.md.erb:9]:
```suggestion
When user impersonation is disabled, PXF executes all Hadoop service requests as the PXF process owner (usually `gpadmin`) or the Hadoop user identity that you specify. This behavior provides no means to control access to Hadoop services for different Greenplum Database users. It requires that this user have access to all files and directories in HDFS, and all tables in Hive and HBase that are referenced in PXF external table definitions.
```

**Review by bradfordb-vmware — APPROVED** [review]

File: docs/content/pxfuserimpers.html.md.erb (hunk 7 above)
Comment (lisakowen) [reply docs/content/pxfuserimpers.html.md.erb:161]: added it back in and included kcd setting.

File: docs/content/pxf_kerbhdfs.html.md.erb [hunk 23]
```diff
@@ -92,7 +130,19 @@ The following table identifies the `pxf.service.user.impersonation` and `pxf.ser
...
-## <a id="procedure"></a>Procedure
+#### Using Kerberos Constrained Delegation
+
+In this configuration, PXF uses Kerberos constrained delegation to request and obtain a ticket on behalf of a \<custom> user, and uses the ticket to access Hadoop.
+
+The following table identifies the impersonation and service user settings, and the PXF and directory service configuration required for this use case; no Hadoop configuration is required:
+
+| Impersonation  | Service&nbsp;User |  PXF Configuration | AD/IPA&nbsp;Config&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; |
+|----------------|-------------------|--------------------|-----------|
+|  false          | \<custom\> | Set up the PXF Kerberos principal, keytab files, and related settings in `pxf-site.xml` as described in the [Configuration Procedure](#procedure) in this topic, disable impersonation as described in [Configure PXF User Impersonation](pxfuserimpers.html#pxf_cfg_impers), [Enable Kerberos Constrained Delegation](#enable_kcd), and [Configure the Hadoop User](pxfuserimpers.html#pxf_cfg_user) to the \<custom> user name.  | Configure `gpadmin` delegation rights for the custom user. |
```
Comment (denalex) [inline docs/content/pxf_kerbhdfs.html.md.erb:141]:
```suggestion
|  false          | \<custom\> | Set up the PXF Kerberos principal, keytab files, and related settings in `pxf-site.xml` as described in the [Configuration Procedure](#procedure) in this topic, disable impersonation as described in [Configure PXF User Impersonation](pxfuserimpers.html#pxf_cfg_impers), [Enable Kerberos Constrained Delegation](#enable_kcd), and [Configure the Hadoop User](pxfuserimpers.html#pxf_cfg_user) to the \<custom> user name.  | Configure AD or IPA to provide the PXF Kerberos principal with the delegation rights for the \<custom> user name|
```

File: docs/content/pxf_kerbhdfs.html.md.erb [hunk 24]
```diff
@@ -274,22 +316,69 @@ When you configure PXF for secure HDFS using an MIT Kerberos KDC server, you wil
...
     PXF automatically replaces ` _HOST` with the FQDN of the host.
 
-6. If you want to access Hadoop as the Greenplum Database user:
+6. [Complete the PXF Configuration](#proc_complete) based on your chosen Hadoop access scenario.
+
+### <a id="proc_complete"></a>Completing the PXF Configuration
 
-    1. Enable user impersonation as described in [Configure PXF User Impersonation](pxfuserimpers.html#pxf_cfg_impers).
-    2. Configure Hadoop proxying for the *primary* component of the Kerberos principal as described in [Configure Hadoop Proxying](pxfuserimpers.html#hadoop). For example, if your principal is `gpadmin/_HOST@EXAMPLE.COM`, configure proxying for the Hadoop user `gpadmin`.
+On the Greenplum Database master host, complete the configuration of the PXF server based on your chosen Hadoop access scenario:
```
Comment (denalex) [inline docs/content/pxf_kerbhdfs.html.md.erb:323]: The options below are mutually exclusive, maybe add a note to this effect so readers understand they should only perform 1 set of actions

**Review by denalex — APPROVED** [review]
Looks pretty good right now. Later, when the time allows, we can revisit this whole security topic and see how we can organize it such that it can be digested easier by readers.

**Comment by lisakowen** [comment]
thank you for all of the reviews!  will follow-up on how this page can be improved.
